boards/components/src/sched/round_robin.rs

```diff
@@ -0,0 +1,74 @@
+//! Component for a round robin scheduler.
+//!
+//! This provides one Component, RoundRobinComponent.
+//!
+//! Usage
+//! -----
+//! (Must be final lines in main.rs)
+//! ```rust
+//! let scheduler = components::round_robin::RoundRobinComponent::new(board_kernel, &PROCESSES)
+//!     .finalize(components::rr_component_helper!(NUM_PROCS));
+//! scheduler.kernel_loop(&imix, chip, Some(&imix.ipc), &main_cap);
```

Comment:
```suggestion
//! Note that `kernel_loop` never returns, thus this must end main.rs
//! ```rust
//! let scheduler = components::round_robin::RoundRobinComponent::new(board_kernel, &PROCESSES)
//!     .finalize(components::rr_component_helper!(NUM_PROCS));
//! scheduler.kernel_loop(&imix, chip, Some(&imix.ipc), &main_cap)
```